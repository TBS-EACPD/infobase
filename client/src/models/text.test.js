import {
  create_text_maker,
  trivial_text_maker,
  run_template,
} from './text.js';

import test_content from './test_content.yaml';

describe("text_maker",()=>{


  describe("run_template",()=>{
    it("should run handlebar templates", () => {
      expect( run_template("{{mar_31}}") ).toEqual("March 31");
      
      
    });
  });

  describe("trival_text_maker",()=> {
    it("should have access to a certain set of global text keys",()=>{
      expect( trivial_text_maker("zoom_out") ).toEqual("Zoom out");
    });
  });

  describe("custom_text_makers",()=>{

    it('should work',()=>{
      const tm_func = create_text_maker(test_content);
      
      const example_with_helpers = tm_func("some_key_with_hbs_helpers", {
        dynamic_key:"other_simple_key",
      })
      expect(example_with_helpers).toMatchSnapshot();
    
      const example_with_hbs_and_md = tm_func("some_key_with_hbs_and_md", {
        show_block_of_text: true,
        html_block_of_text: `
          This should show up twice, once <strong>escaped</strong> and once <span> unescaped </span>
          <div style="width:100px;height:100px;background-color:red;"></div>
        `,
        list: ["a", "b", "c" ],
      });
      expect(example_with_hbs_and_md).toMatchSnapshot();
    })

  })
})
