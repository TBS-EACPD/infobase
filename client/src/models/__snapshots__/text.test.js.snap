// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text_maker custom_text_makers should work 1`] = `
"this key uses another: some_simple_key: A simple key
also uses dynamic key: other_simple_key: another simple key"
`;

exports[`text_maker custom_text_makers should work 2`] = `
"<h1 id=\\"title\\">title</h1>
<p>some text with <em>italics</em> and <strong>bold</strong></p>
<p>  the following block should be unescaped:
  <div></p>
<pre><code>      This should show up twice, once &amp;lt;strong&amp;gt;escaped&amp;lt;/strong&amp;gt; and once &amp;lt;span&amp;gt; unescaped &amp;lt;/span&amp;gt;
      &amp;lt;div style&amp;#x3D;&amp;quot;width:100px;height:100px;background-color:red;&amp;quot;&amp;gt;&amp;lt;/div&amp;gt;</code></pre>  </div>

<p>  the following block should be unescaped:</p>
  <div>

<pre><code>      This should show up twice, once &lt;strong&gt;escaped&lt;/strong&gt; and once &lt;span&gt; unescaped &lt;/span&gt;
      &lt;div style=&quot;width:100px;height:100px;background-color:red;&quot;&gt;&lt;/div&gt;</code></pre>  </div>


<p>list below</p>
<ul>
<li>a</li>
<li>b</li>
<li>c</li>
</ul>
"
`;
